File: Python/Calcular_IMC.py
print("CALCULADORA DE IMC")
print("=" * 50)

nombre = input("Ingrese su nombre: ").strip()
edad = int(input("Ingrese su edad (1-120 años): "))
peso = float(input("Ingrese su peso en kg (20-300): "))
altura_cm = float(input("Ingrese su altura en centímetros (50-250): "))

# Validaciones
if edad < 1 or edad > 120:
    print("Advertencia: Edad fuera del rango recomendado (1-120).")

if peso < 20 or peso > 300:
    print("Advertencia: Peso fuera del rango recomendado (20-300 kg).")

if altura_cm < 50 or altura_cm > 250:
    print("Advertencia: Altura fuera del rango recomendado (50-250 cm).")

# Convertir altura a metros
altura_m = altura_cm / 100.0

# Calcular IMC
imc = peso / (altura_m ** 2)

if imc < 18.5:
    diagnostico = "Bajo de peso"
elif 18.5 <= imc < 24.9:
    diagnostico = "Peso normal"
elif 25 <= imc < 29.9:
    diagnostico = "Sobrepeso"
elif 30 <= imc < 34.9:
    diagnostico = "Obesidad tipo 1"
elif 35 <= imc < 39.9:
    diagnostico = "Obesidad tipo 2"
else:
    diagnostico = "Obesidad tipo 3 (obesidad mórbida)"

# Resultado - Formato solicitado
print("\n" + "=" * 40)
print("     INFORME DE DIAGNÓSTICO IMC")
print("=" * 40)
print(f"Paciente: {nombre}")
print(f"Edad: {edad} años")
print(f"Peso: {peso:.2f} kg")
print(f"Altura: {altura_m:.2f} m (convertido de {altura_cm:.1f} cm)")
print(f"\nRESULTADOS:")
print(f"IMC calculado: {imc:.2f}")
print(f"Diagnóstico: {diagnostico}")
print("=" * 40)

File: C/Calcular_IMC.c
#include <stdio.h>
#include <string.h>

char nombre[50];
int edad;
float peso, altura_cm, altura_m, imc;
char diagnostico[50];

int main() {
    printf("Ingrese su nombre: ");
    fgets(nombre, 50, stdin);
    nombre[strcspn(nombre, "\n")] = 0;
    printf("Ingrese su edad (1-120 años): ");
    scanf("%d", &edad);
    printf("Ingrese su peso en kg (20-300): ");
    scanf("%f", &peso);
    printf("Ingrese su altura en centímetros (50-250): ");
    scanf("%f", &altura_cm);
    
    if (edad < 1 || edad > 120) {
        printf("Advertencia: Edad fuera del rango recomendado (1-120).\n");
    }
    
    if (peso < 20 || peso > 300) {
        printf("Advertencia: Peso fuera del rango recomendado (20-300 kg).\n");
    }
    
    if (altura_cm < 50 || altura_cm > 250) {
        printf("Advertencia: Altura fuera del rango recomendado (50-250 cm).\n");
    }

    // Convertir altura a metros
    altura_m = altura_cm / 100.0;
    
    // Calcular IMC
    imc = peso / (altura_m * altura_m);
    
    // Determinar diagnóstico usando if-else if-else
    if (imc < 18.5) {
        strcpy(diagnostico, "Bajo de peso");
    } else if (imc >= 18.5 && imc < 24.9) {
        strcpy(diagnostico, "Peso normal");
    } else if (imc >= 25 && imc < 29.9) {
        strcpy(diagnostico, "Sobrepeso");
    } else if (imc >= 30 && imc < 34.9) {
        strcpy(diagnostico, "Obesidad tipo 1");
    } else if (imc >= 35 && imc < 39.9) {
        strcpy(diagnostico, "Obesidad tipo 2");
    } else {
        strcpy(diagnostico, "Obesidad tipo 3 (obesidad mórbida)");
    }
    
    // Resultado - Formato solicitado
    printf("\n\n");
    printf("========================================\n");
    printf("     INFORME DE DIAGNÓSTICO IMC\n");
    printf("========================================\n");
    printf("Paciente: %s\n", nombre);
    printf("Edad: %d años\n", edad);
    printf("Peso: %.2f kg\n", peso);
    printf("Altura: %.2f m (convertido de %.1f cm)\n", altura_m, altura_cm);
    printf("\nRESULTADOS:\n");
    printf("IMC calculado: %.2f\n", imc);
    printf("Diagnóstico: %s\n", diagnostico);
    printf("========================================\n");
    
    return 0;
}